core/src/main/scala/tech/beshu/ror/boot/RorInstance.scala

```diff
@@ -112,72 +122,94 @@ class RorInstance private(boot: ReadonlyRest,
   }
 
   def forceReloadTestConfigEngine(config: RawRorConfig,
-                                  ttl: FiniteDuration)
-                                 (implicit requestId: RequestId): Task[Either[RawConfigReloadError, Unit]] = {
+                                  ttl: FiniteDuration Refined Positive)
+                                 (implicit requestId: RequestId): Task[Either[IndexConfigReloadWithUpdateError, Unit]] = {
     anTestConfigEngine.forceReloadTestConfigEngine(config, ttl)
   }
 
   def invalidateTestConfigEngine()
-                                (implicit requestId: RequestId): Task[Unit] = {
+                                (implicit requestId: RequestId): Task[Either[IndexConfigInvalidationError, Unit]] = {
     anTestConfigEngine.invalidateTestConfigEngine()
   }
 
   def stop(): Task[Unit] = {
     implicit val requestId: RequestId = RequestId("ES sigterm")
     for {
+      _ <- Task.delay(configsReloadTask.cancel())
       _ <- anTestConfigEngine.stop()
       _ <- aMainConfigEngine.stop()
     } yield ()
   }
 
-  private def scheduleIndexConfigChecking(interval: FiniteDuration Refined Positive): Cancelable = {
-    logger.debug(s"[CLUSTERWIDE SETTINGS] Scheduling next in-index settings check within $interval")
+  private def scheduleMainConfigReload(interval: FiniteDuration Refined Positive) = {
+    scheduleIndexConfigChecking(interval, "main", tryMainEngineReload)
+  }
+
+  private def scheduleTestConfigReload(interval: FiniteDuration Refined Positive) = {
+    scheduleIndexConfigChecking(interval, "test", tryTestEngineReload)
+  }
+
+  private def scheduleIndexConfigChecking(interval: FiniteDuration Refined Positive,
+                                          name: String,
+                                          reloadTask: RequestId => Task[Either[ScheduledReloadError, Unit]]): Cancelable = {
+    logger.debug(s"[CLUSTERWIDE SETTINGS] Scheduling next in-index $name settings check within ${interval.value}")
```

Comment:
let's stop here. 

Currently, we have two engines and two scheduled tasks.

Did you consider one scheduled task that tries to reload eg. a list of engines? What approach is better and why? WDYT?

Reply:
Finally, I used a list of engines